debugger/memory_maps.md

```diff
@@ -0,0 +1,161 @@
+    0x7fa80a19d000 /usr/lib/ld-2.27.so
+
+Now we can see that `libc-2.27.so` was loaded as well, great!
+
+Speaking of `libc`, a popular task for binary exploitation is to find the address of a specific symbol in a library. With this information in hand, you can build, for example, an exploit which uses ROP. This can be achieved using the `dmi` command. So if we want, for example, to find the address of [`system()`](http://man7.org/linux/man-pages/man3/system.3.html) in the loaded `libc`, we can simply execute the following command:
+
+    [0x55ca23a4a520]> dmi libc system
+    514 0x00000000 0x7fa809de1000  LOCAL   FILE    0 system.c
+    515 0x00043750 0x7fa809e24750  LOCAL   FUNC 1221 do_system
+    4468 0x001285a0 0x7fa809f095a0  LOCAL   FUNC  100 svcerr_systemerr
+    5841 0x001285a0 0x7fa809f095a0  LOCAL   FUNC  100 svcerr_systemerr
+    6427 0x00043d10 0x7fa809e24d10   WEAK   FUNC   45 system
+    7094 0x00043d10 0x7fa809e24d10 GLOBAL   FUNC   45 system
+    7480 0x001285a0 0x7fa809f095a0 GLOBAL   FUNC  100 svcerr_systemerr
+
+Similar to the `dm.` command, with `dmi.` you can see the closest symbol to the current address.
+
+Another useful command is to list the sections of a specific library. In the following example we'll list the sections of `ld-2.27.so`:
+
+    [0x55a7ebf09520]> dmS ld-2.27
+    [Sections]
+    00 0x00000000     0 0x00000000     0 ---- ld-2.27.so.
+    01 0x000001c8    36 0x7fae4652d1c8    36 -r-- ld-2.27.so..note.gnu.build_id
+    02 0x000001f0   352 0x7fae4652d1f0   352 -r-- ld-2.27.so..hash
+    03 0x00000350   412 0x7fae4652d350   412 -r-- ld-2.27.so..gnu.hash
+    04 0x000004f0   816 0x7fae4652d4f0   816 -r-- ld-2.27.so..dynsym
+    05 0x00000820   548 0x7fae4652d820   548 -r-- ld-2.27.so..dynstr
+    06 0x00000a44    68 0x7fae4652da44    68 -r-- ld-2.27.so..gnu.version
+    07 0x00000a88   164 0x7fae4652da88   164 -r-- ld-2.27.so..gnu.version_d
+    08 0x00000b30  1152 0x7fae4652db30  1152 -r-- ld-2.27.so..rela.dyn
+    09 0x00000fb0 114976 0x7fae4652dfb0 114976 -r-x ld-2.27.so..text
+    10 0x0001d0e0 17760 0x7fae4654a0e0 17760 -r-- ld-2.27.so..rodata
+    11 0x00021640  1716 0x7fae4654e640  1716 -r-- ld-2.27.so..eh_frame_hdr
+    12 0x00021cf8  9876 0x7fae4654ecf8  9876 -r-- ld-2.27.so..eh_frame
+    13 0x00024660  2020 0x7fae46751660  2020 -rw- ld-2.27.so..data.rel.ro
+    14 0x00024e48   336 0x7fae46751e48   336 -rw- ld-2.27.so..dynamic
+    15 0x00024f98    96 0x7fae46751f98    96 -rw- ld-2.27.so..got
+    16 0x00025000  3960 0x7fae46752000  3960 -rw- ld-2.27.so..data
+    17 0x00025f78     0 0x7fae46752f80   376 -rw- ld-2.27.so..bss
+    18 0x00025f78    17 0x00000000    17 ---- ld-2.27.so..comment
+    19 0x00025fa0    63 0x00000000    63 ---- ld-2.27.so..gnu.warning.llseek
+    20 0x00025fe0 13272 0x00000000 13272 ---- ld-2.27.so..symtab
+    21 0x000293b8  7101 0x00000000  7101 ---- ld-2.27.so..strtab
+    22 0x0002af75   215 0x00000000   215 ---- ld-2.27.so..shstrtab
+
+## Memory map of the Heap
```

Comment:
put that in a different file, otherwise the index doesnt looks good. debugger/heap.md

Reply:
fixed